csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -131,6 +132,40 @@ public DependencyManager(string srcDir, IDependencyOptions options, ILogger logg
                 progressMonitor.UnresolvedReference(r.Key, r.Value);
             }
 
+            var webViewExtractionOption = Environment.GetEnvironmentVariable("CODEQL_EXTRACTOR_CSHARP_STANDALONE_EXTRACT_WEB_VIEWS");
+            if (bool.TryParse(webViewExtractionOption, out var shouldExtractWebViews) &&
+                shouldExtractWebViews)
+            {
+                progressMonitor.LogInfo($"Generating source files from cshtml and razor files.");
+
+                var views = GetFiles("*.cshtml")
+                    .Concat(GetFiles("*.razor"))
+                    .ToArray();
+
+                if (views.Length > 0)
+                {
+                    progressMonitor.LogInfo($"Found {views.Length} cshtml and razor files.");
+
+                    // TODO: use SDK specified in global.json
+                    var sdk = new Sdk(dotnet).GetNewestSdk();
+                    if (sdk != null)
+                    {
+                        try
+                        {
+                            var razor = new Razor(sdk, dotnet, progressMonitor);
+                            razorWorkingDirectory = new TemporaryDirectory(ComputeTempDirectory(sourceDir.FullName, "razor"));
+                            var generatedFiles = razor.GenerateFiles(views, usedReferences.Keys, razorWorkingDirectory.ToString());
```

Comment:
nit: consider to use `razoeWorkingDirectory.FullName` instead of calling `ToString()`.

Reply:
Just realized that `TemporaryDirectory` has no `FullName`. `ToString()` calls the underlying `DirectoryInfo.FullName`.